### DISCUSSION TITLE
Assign Null and Schema design

### DISCUSSION THREAD

**Discussion by Tokenyet** [typegoose/typegoose#625] [Q&A]
Just wondering how to assign null to a field, so I find out the code in the [typeguards.ts](https://github.com/typegoose/typegoose/blob/aaccd0e812c14ec2beb1ad3e8091577066fd63aa/test/models/typeguards.ts).

```
export class UserRef {
  @prop({ ref: UserRef, default: null })
  public master?: Ref<UserRef>;

  @prop({ ref: UserRef, default: [] })
  public subAccounts?: Ref<UserRef>[];

  @prop({ required: true })
  public name!: string;
}
```

So master is `nullable` according to `@prop({ ref: UserRef, default: null })`, but should we use `Nullable<Ref<UserRef>>` instead of `Ref<UserRef>`?

I know ` public master?: Ref<UserRef>;` is equivalent to `public master: Ref<UserRef> | undefined;` but not `null`, so just wondering If `Nullable<Ref<UserRef>>` or `Ref<UserRef> | null` is more type-correct?

**Comment by hasezoey** [comment]
i dont know your specifications, but i would recommend to use `default: undefined` and `public master?: Ref<UserRef>;` over `null`
but if really required, then the correct type would be `Ref<UserRef> | null` or `Nullable<Ref<UserRef>>` should also be correct (both of which are not tested in typegoose, so i dont know if it will actually compile)
